### PR TITLE
Fix "map does not respect sources index type" (#350)

### PR DESCRIPTION
**Pull request by dpinol**

### CODE REVIEW

**Comment by SobhanMP** [comment]
that's a nice catch!

File: src/sparsevector.jl [hunk 1]
```diff
@@ -1468,7 +1469,7 @@ function _binarymap_mode_1!(f::Function, mx::Int, my::Int,
     return ir
 end
 
-function _binarymap_mode_2!(f::Function, mx::Int, my::Int,
+function _binarymap_mode_2!(f::Function, mx::Integer, my::Integer,
```
Comment (SobhanMP) [inline src/sparsevector.jl:1472]: but now `ix` will not have the same type as `mx`, i think the safe thing is just make every function take Int and then store the "right" type. `Int` seems to be fairly standard for indexing in julia IMHO.

Comment (SobhanMP) [reply src/sparsevector.jl:1472]: My suggestion would be to undo all changes after line 1395, unless we test for type stability.

Comment (dpinol) [reply src/sparsevector.jl]: Ok, I undid changes after line 1395 and rebased my commit to keep history clean

**Review by Wimmerer — APPROVED** [review]

**Pull request merged**